src/resdk/query.py

```diff
@@ -380,3 +388,52 @@ def iterate(self, chunk_size=100, show_progress=False):
                     min_id = obj.id
                     pbar.update(1)
                     yield obj
+
+
+class AnnotationFieldQuery(ResolweQuery):
+    """Add additional method to the annotation field query."""
+
+    def from_path(self, full_path: str) -> "AnnotationField":
+        """Get the AnnotationField from full path.
+
+        :raises LookupError: when field at the specified path does not exist.
+        """
+        group_name, field_name = full_path.split(".", maxsplit=1)
+        return self.get(name=field_name, group__name=group_name)
+
+
+class AnnotationValueQuery(ResolweQuery):
+    """Populate Annotation fields with a single query."""
+
+    def _fetch(self):
+        """Make request to the server and populate cache.
+
+        Fetch all values and their fields with 2 queries.
+        """
+        # Execute the query in a single request.
+        super()._fetch()
+
+        missing = {
```

Comment:
Cold you explain why this scenario happens? Why in some cases `self._field` is None, while in others it is not None?

Reply:
This is only optimization: the annotation fields are fetched in a single query when reading corresponding annotation values.

Maybe this is not necessary. Consider the following use case:

```
values = res.annotation_values.filter(entity=xxx)
for value in values:
    print(value.field.name, value.field.group.name)
```

This would query the backend for every field (so it would produce n queries). The "optimized" fetch only produces two queries (one for values and one for fields). One could argue that fields are not always used so the second query is not always necessary.